hiss/team/views.py

```diff
@@ -37,14 +43,19 @@ def form_valid(self, form: CreateTeamForm):
         return redirect(team.get_absolute_url())
 
 
-class JoinTeamView(shared_mixins.LoginRequiredAndAppliedMixin, generic.FormView):
+class JoinTeamView(shared_mixins.UserHasNoTeamMixin, generic.FormView):
     """
     Adds the user to a team (if the team isn't already at capacity).
     """
 
     form_class = JoinTeamForm
     template_name = "team/join_form.html"
 
+    def handle_no_permission(self):
+        if not self.request.user.is_authenticated:
+            return redirect("/accounts/login/?next=/team/join")
```

Comment:
Please use `reverse_lazy` here instead of hard-coding the URL.

```suggestion
            return redirect(f"{reverse_lazy('customauth:login')}?next={reverse_lazy('team:join')}")
```